src/plugins/elasticsearch/index.js

```diff
@@ -1,38 +1,38 @@
-var _ = require('lodash');
-let Boom = require('boom');
+import { trim, trimRight } from 'lodash';
+import { methodNotAllowed } from 'boom';
 
-module.exports = function (kibana) {
-  var healthCheck = require('./lib/health_check');
-  var exposeClient = require('./lib/expose_client');
-  var createProxy = require('./lib/create_proxy');
+import healthCheck from './lib/health_check';
+import exposeClient from './lib/expose_client';
+import createProxy, { createPath } from './lib/create_proxy';
 
-  return new kibana.Plugin({
+module.exports = function ({ Plugin }) {
```

Comment:
The reason this change was made in babel 6 is that babel 5 allowed broken code, e.g.

```
export default {
  foo: 1,
  bar: 2
}

// then somewhere

import { foo } from './above-file'; // BOOM
```

This is not allowed according to the spec (and babel 6), but is allowed in Babel 5.

At some point we definitely should upgrade, otherwise we won't get the benefits of everything happening with babel 6 (and later versions) — fewer people on babel 5 means fewer bug fixes, fewer TC39 proposals as plugins etc. There's already ~1100 commits between 5.x and master. If we want to have a shared build system I think it's important to keep it up-to-date.

The problem is that Kibana mixes `import/export` and `require/module.exports`, which we perhaps shouldn't do. Maybe try to use a codemod to transform the code? I ran https://gist.github.com/ide/f3266c3915b0e45dac78 on `src/ui` and `src/plugins` to make most `require`s become `import`s. All tests still ran, and everything seemed to work. It still doesn't handle `requirefrom` usage of course. And it doesn't handle the exports. I just tested to see how it worked. (We could also go the other way and only rely on `require/module.exports`)